wormhole-connect/src/hooks/useBalanceChecker.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState, useMemo } from 'react';
+import { amount, routes } from '@wormhole-foundation/sdk';
+import { getWalletConnection, TransferWallet } from 'utils/wallet';
+import config from 'config';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
+  isCheckingBalance: boolean;
+  hasSufficientBalance: boolean;
+  walletBalance?: number;
+  networkCost?: number;
+} => {
+  const [isCheckingBalance, setCheckingBalance] = useState(false);
+  const [state, setState] = useState<{ balance: number; cost: number } | null>(
+    null,
+  );
+
+  const feeSymbol = useMemo(() => {
+    if (!quote?.relayFee?.token) return;
+
+    return config.sdkConverter.findTokenConfigV1(
+      quote?.relayFee?.token,
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    let isActive = true;
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        if (!quote?.relayFee?.amount) return;
+
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet) return;
+
+        const cost = amount.whole(quote.relayFee.amount);
+
+        const feeTokenAddress = quote.relayFee.token.address?.toString();
```

Comment:
If token.chain != fromChain we should bail out

Reply:
added that change